prow/bugzilla/client.go

```diff
@@ -509,6 +510,86 @@ func (c *client) AddPullRequestAsExternalBug(id int, org, repo string, num int)
 	return changed, nil
 }
 
+// RemovePullRequestAsExternalBug attempts to remove a PR from the external tracker list.
+// External bugs are assumed to fall under the type identified by their hostname,
+// so we will provide https://github.com/ here for the URL identifier. We return
+// any error as well as whether a change was actually made.
+// This will be done via JSONRPC:
+// https://bugzilla.redhat.com/docs/en/html/integrating/api/Bugzilla/Extension/ExternalBugs/WebService.html#remove-external-bug
+func (c *client) RemovePullRequestAsExternalBug(id int, org, repo string, num int) (bool, error) {
+	logger := c.logger.WithFields(logrus.Fields{methodField: "RemoveExternalBug", "id": id, "org": org, "repo": repo, "num": num})
+	pullIdentifier := IdentifierForPull(org, repo, num)
+	rpcPayload := struct {
+		// Version is the version of JSONRPC to use. All Bugzilla servers
+		// support 1.0. Some support 1.1 and some support 2.0
+		Version string `json:"jsonrpc"`
+		Method  string `json:"method"`
+		// Parameters must be specified in JSONRPC 1.0 as a structure in the first
+		// index of this slice
+		Parameters []RemoveExternalBugParameters `json:"params"`
+		ID         string                        `json:"id"`
+	}{
+		Version: "1.0", // some Bugzilla servers support 2.0 but all support 1.0
+		Method:  "ExternalBugs.remove_external_bug",
+		ID:      "identifier", // this is useful when fielding asynchronous responses, but not here
+		Parameters: []RemoveExternalBugParameters{{
+			APIKey: string(c.getAPIKey()),
+			BugIDs: []int{id},
+			ExternalBugIdentifier: ExternalBugIdentifier{
+				Type: "https://github.com/",
+				ID:   pullIdentifier,
+			},
+		}},
+	}
+	body, err := json.Marshal(rpcPayload)
+	if err != nil {
+		return false, fmt.Errorf("failed to marshal JSONRPC payload: %v", err)
+	}
+	req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/jsonrpc.cgi", c.endpoint), bytes.NewBuffer(body))
+	if err != nil {
+		return false, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.request(req, logger)
+	if err != nil {
+		return false, err
+	}
+	var response struct {
+		Error *struct {
+			Code    int    `json:"code"`
+			Message string `json:"message"`
+		} `json:"error,omitempty"`
+		ID     string `json:"id"`
+		Result *struct {
+			ExternalBugs []struct {
+				Type string `json:"ext_type_url"`
+				ID   string `json:"ext_bz_bug_id"`
+			} `json:"external_bugs"`
+		} `json:"result,omitempty"`
+	}
+	if err := json.Unmarshal(resp, &response); err != nil {
+		return false, fmt.Errorf("failed to unmarshal JSONRPC response: %v", err)
+	}
+	if response.Error != nil {
+		if response.Error.Code == 1006 && strings.Contains(response.Error.Message, `duplicate key value violates unique constraint "ext_bz_bug_map_bug_id_idx"`) {
```

Comment:
Regarding [remove-external-bug docs](https://bugzilla.redhat.com/docs/en/html/integrating/api/Bugzilla/Extension/ExternalBugs/WebService.html#remove-external-bug), `1006` error code means that `No external tracker bugs were found that matched your criteria`. What is the reason of checking what the error message contains?